aws-iot-accountauditconfiguration/src/main/java/com/amazonaws/iot/accountauditconfiguration/CreateHandler.java

```diff
@@ -0,0 +1,163 @@
+            ResourceHandlerRequest<ResourceModel> request,
+            CallbackContext callbackContext,
+            Logger logger) {
+
+        ResourceModel model = request.getDesiredResourceState();
+
+        // We ask customers to specify the Account ID as part of the model,
+        // because there must be a primary identifier.
+        // Having it in the model helps CFN prevent cases like 2 stacks from managing the same resource.
+        String accountIdFromTemplate = model.getAccountId();
+        String accountId = request.getAwsAccountId();
+        if (!accountIdFromTemplate.equals(accountId)) {
+            return ProgressEvent.failed(model, callbackContext, HandlerErrorCode.InvalidRequest,
+                    String.format("AccountId in the template (%s) doesn't match actual: %s.",
+                            accountIdFromTemplate, accountId));
+        }
+
+        // Call Describe to see whether the customer has a configuration already.
+        // Note that this API never throws ResourceNotFoundException.
+        // If the configuration doesn't exist, it returns an object with all checks disabled,
+        // other fields equal to null.
+        DescribeAccountAuditConfigurationResponse describeResponse;
+        try {
+            describeResponse = proxy.injectCredentialsAndInvokeV2(
+                    DescribeAccountAuditConfigurationRequest.builder().build(),
+                    iotClient::describeAccountAuditConfiguration);
+        } catch (IotException e) {
+            throw Translator.translateIotExceptionToCfn(e);
+        }
+        logger.log("Called DescribeAccountAuditConfiguration for " + accountId);
+
+        // We judge whether the configuration exists by the RoleArn field.
+        // A customer cannot create a configuration without specifying a RoleArn.
+        // For an existing configuration, the RoleArn can never be nullified,
+        // unless the whole configuration is deleted.
+        boolean roleArnAlreadyExists = !StringUtils.isEmpty(describeResponse.roleArn());
+        if (roleArnAlreadyExists) {
+            // Note: we don't fail with AlreadyExists as soon as we see an existing configuration.
+            // We return success if the existing configuration is identical.
+            // We do this to not break the customer's template in case of a transient exception.
+            // Consider a scenario where our CreateHandler succeeds, but our response doesn't reach CFN.
+            // CFN would retry. If we returned AlreadyExists, CFN would throw "CREATE_FAILED: resource already exists"
+            // to the customer.
+            // We do know that this configuration could've been created manually rather than through
+            // CFN, but, unlike for other resources, we don't have a way to tell how
+            // the configuration was created.
+            boolean areEquivalent = areEquivalent(model, describeResponse, logger);
+            if (areEquivalent) {
```

Comment:
Maybe we should add some logs to show which path we entered.

Reply:
Sounds good, added new logs.